cliquet/resource/__init__.py

```diff
@@ -240,6 +248,11 @@ def collection_get(self):
             next_page = self._next_page_url(sorting, limit, lastrecord, offset)
             headers['Next-Page'] = encode_header(next_page)
 
+        records = [
+            {field: val for field, val in record.items() if field in fields}
```

Comment:
What if a *known field* is not present in the record ?

I would suggest to let extract_fields return `None` and don't perform the filtering if no `?_fields=` is specified (for performance reason also)